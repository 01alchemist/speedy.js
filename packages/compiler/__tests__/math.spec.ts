import {runCases} from "./test-case-runner";

runCases("Math", "math");
