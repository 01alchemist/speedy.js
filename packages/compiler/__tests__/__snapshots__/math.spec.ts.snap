// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Math math.sqrt 1`] = `
"; ModuleID = '__tests__/cases/math/math.sqrt.ts'
source_filename = \\"__tests__/cases/math/math.sqrt.ts\\"
target datalayout = \\"e-m:e-p:32:32-i64:64-n32:64-S128\\"
target triple = \\"wasm32-unknown-unknown\\"

@Array_name = private unnamed_addr constant [6 x i8] c\\"Array\\\\00\\"
@Array_type_descriptor = private constant { [6 x i8]* } { [6 x i8]* @Array_name }
@Math_name = private unnamed_addr constant [5 x i8] c\\"Math\\\\00\\"
@Math_type_descriptor = private constant { [5 x i8]* } { [5 x i8]* @Math_name }
@Math_object = private constant { { [5 x i8]* }* } { { [5 x i8]* }* @Math_type_descriptor }
@Math_ptr = private constant { { [5 x i8]* }* }* @Math_object

define void @mathSqrt() {
entry:
  %intValue = alloca i32
  %mathPtr = load { { [5 x i8]* }* }*, { { [5 x i8]* }* }** @Math_ptr, align 4
  %0 = call double @llvm.sqrt.f64(double 3.140000e+00)
  store i32 4, i32* %intValue, align 4
  %mathPtr1 = load { { [5 x i8]* }* }*, { { [5 x i8]* }* }** @Math_ptr, align 4
  %intValue2 = load i32, i32* %intValue, align 4
  %1 = sitofp i32 %intValue2 to double
  %2 = call double @llvm.sqrt.f64(double %1)
  ret void
}

; Function Attrs: nounwind readnone
declare double @llvm.sqrt.f64(double) #0

attributes #0 = { nounwind readnone }
"
`;
